_posts/2018-06-13-rpm-ostree-automatic-updates.md

```diff
@@ -0,0 +1,178 @@
+---
+layout: post
+title: Using Automatic Updates on Fedora Silverblue
+---
+
+One of the [oldest issues](https://github.com/projectatomic/rpm-ostree/issues/247)
+and most desired pieces of functionality in the `rpm-ostree` stack has been
+automatic updates.  The ability to have your host download and stage the
+upgrade in the background would eliminate some of the system admin headache
+that comes when maintaining your host.  (Furthermore, enabling this ability
+in a cluster of hosts allows the cluster admin to worry about other things
+than scheduling downtime to apply upgrades.)
+
+Earlier this year, the `rpm-ostree` developers made the first step towards the
+ultimate goal of automated updates, when the ability to [check for updates](https://github.com/projectatomic/rpm-ostree/pull/1147)
+in the background landed in the codebase.  The background checking and the
+[nicely formatted status](https://github.com/projectatomic/rpm-ostree/pull/1147#issuecomment-355614658)
+was a welcome addition to the experience of administering a host with `rpm-ostree`.
+
+In version [v2018.5](https://github.com/projectatomic/rpm-ostree/releases/tag/v2018.5)
+of `rpm-ostree`, the next piece of the puzzle landed.  `rpm-ostree` gained the
+ability to [download and stage the upgrade in the background](https://github.com/projectatomic/rpm-ostree/pull/1321).
+Not only did this further enable automatic updates, but it also allowed the
+developers to address another [long standing issue](https://github.com/projectatomic/rpm-ostree/issues/40)
+about performing the `/etc/` merge just before reboot, rather than at the time
+of the upgrade.
+
+In this post, I'll cover how to enable the various automatic update policies
+that are now available in the `rpm-ostree` stack.
+
+## Automatic Update Checking
+
+As mentioned earlier, the first step towards complete automatic updates was
+enabling the checking of updates in the background.  This is done using the
+`/etc/rpm-ostreed.conf` config file, a systemd timer, and systemd service.
+
+Below you can see the output from `rpm-ostree status` before any of the
+update policies have been configured. The `State:` field describes that
+the `rpm-ostreed` daemon is idle and automatic updates are disabled.
+
+```
+# rpm-ostree status
+State: idle; auto updates disabled
+Deployments:
+● ostree://fedora-workstation:fedora/28/x86_64/workstation
+                   Version: 28.20180601.0 (2018-06-01 11:53:40)
+                    Commit: 6067049927936b5bf41695263c43f37238603b858506a45c98316501c94841b0
+              GPGSignature: Valid signature by 128CF232A9371991C8A65695E08E7E629DB62FB1
+```
+
+We can turn on the update checking by editing `/etc/rpm-ostreed.conf`
+and uncommenting the `AutomaticUpdatePolicy` and `IdleExitTimeout`
```

Comment:
The `IdleExitTimeout` is not related to automatic updates, so you don't have to necessarily uncomment/do anything special with it.